src/v/utils/fragmented_vector.h

```diff
@@ -222,6 +222,57 @@ class fragmented_vector {
         _capacity = 0;
     }
 
+    /**
+     * Return the number of elements the vector has allocated space for.
+     */
+    size_t capacity() const { return _capacity; }
+
+    /**
+     * Resizes the vector to contain \p count elements. If the current size is
+     * less than \p count then default-initialized elements are added. If the
+     * current size is larger then elements are removed.
+     *
+     * The expected use case for this is to allocate a large vector in a fiber
+     * using a series of smaller resize() invocations allowing for cooperative
+     * yield calls to be inserted to avoid reactor stalls. The optimal strategy
+     * would be to fill the latest fragment by inspecting the current capacity()
+     * and then to make allocations in increments of elements_per_fragment().
+     */
+    void resize(const size_t count) {
+        if (_size == count) {
+            return;
+        }
+
+        if (_size > count) {
+            pop_back_n(size() - count);
+            return;
+        }
+
+        auto deficit = count - _size;
+        while (deficit) {
+            if (_size == _capacity) {
+                std::vector<T> frag;
+                if (deficit >= elems_per_frag) {
+                    frag.resize(elems_per_frag);
```

Comment:
nit: not sure if `std::vector::resize` has a guarantee that `capacity == size` after a call which requires increasing capacity

Reply:
Could you expand on the concern you have? Is the concern that `capacity != size` related a correctness concern, or an optimization concern (i guess or both)?

Reply:
My concern is that if `resize` tries to be smart and increases the capacity more than strictly required, then `_capacity` would become out of sync with the actual capacity. Played around with this and it doesn't seem to be the case though.

Reply:
>capacity more than strictly required, then _capacity would become out of sync

got it. i don't think we ever inspect the capacity of the underlying vectors, so it sounds like your concern is optimization rather than out-of-sync affecting correctness?

>My concern is that if resize tries to be smart and increases the capacity more than strictly required

Was just reading the implementation in libc++. It does seem to double the capacity when resize() appends new entries. However, since that adaptive capacity feature operates on capacity and capacity of a new vector is zero, then resize() on a empty vector does the right thing and uses a precise allocation size.

Interesting!

Reply:
Yeah, I don't think it impacts correctness, but would be surprising. I think you can always do a `shrink_to_fit` call after the resize, but maybe it doesn't matter that much.

Reply:
I just added a reserve() and a note. In this case, at least in libc++ resize won't do any new allocations if the capacity is sufficient.

Reply:
I ended up removing frag_vec::resize i think it was a bit awkward